java/arcs/android/storage/database/DatabaseImpl.kt

```diff
@@ -1076,22 +1127,26 @@ class DatabaseImpl(
         suspend fun insertFieldBlock(
             fieldName: String,
             fieldType: FieldType,
-            isCollection: Boolean
+            isCollection: FieldClass
         ) {
             counters?.increment(DatabaseCounters.INSERT_ENTITY_FIELD)
             insertFieldStatement.apply {
                 bindLong(1, getTypeId(fieldType, db))
                 bindLong(2, schemaTypeId)
                 bindString(3, fieldName)
-                bindBoolean(4, isCollection)
+                bindLong(4, isCollection.ordinal.toLong())
                 executeInsert()
             }
         }
         schema.fields.singletons.forEach { (fieldName, fieldType) ->
-            insertFieldBlock(fieldName, fieldType, isCollection = false)
+            val isCollection = when {
+                fieldType.tag == FieldType.Tag.List -> FieldClass.List
+                else -> FieldClass.Singleton
+            }
```

Comment:
would this be more readable with an if?
val isCollection = if(fieldType.tag == FieldType.Tag.List) FieldClass.List else FieldClass.Singleton